crates/scroll/evm/src/execute.rs

```diff
@@ -0,0 +1,544 @@
+    }
+}
+
+impl<DB, EvmConfig> BlockExecutionStrategy<DB> for ScrollExecutionStrategy<DB, EvmConfig>
+where
+    DB: Database<Error: Into<ProviderError> + Display>,
+    State<DB>: FinalizeExecution<Output = BundleState>,
+    EvmConfig: ConfigureEvm<Header = Header>,
+{
+    type Error = BlockExecutionError;
+
+    fn apply_pre_execution_changes(
+        &mut self,
+        block: &BlockWithSenders,
+        _total_difficulty: U256,
+    ) -> Result<(), Self::Error> {
+        // TODO (scroll): update to the Scroll chain spec
+        // TODO (scroll): update to the Curie hardfork
+        if self.chain_spec.fork(EthereumHardfork::Dao).transitions_at_block(block.number) {
+            if let Err(err) = apply_curie_hard_fork(&mut self.state) {
+                tracing::debug!(%err, "failed to apply curie hardfork");
+                return Err(ForkError::Curie.into());
+            };
+        }
+
+        Ok(())
+    }
+
+    fn execute_transactions(
+        &mut self,
+        block: &BlockWithSenders,
+        total_difficulty: U256,
+    ) -> Result<ExecuteOutput, Self::Error> {
+        let env = self.evm_env_for_block(&block.header, total_difficulty);
+        let mut evm = self.evm_config.evm_with_env(&mut self.state, env);
+
+        let mut cumulative_gas_used = 0;
+        let mut receipts = Vec::with_capacity(block.body.transactions.len());
+
+        for (sender, transaction) in block.transactions_with_sender() {
+            // The sum of the transaction’s gas limit and the gas utilized in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = block.header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                return Err(BlockValidationError::TransactionGasLimitMoreThanAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                }
+                .into())
+            }
+
+            if transaction.is_eip4844() {
+                return Err(ConsensusError::InvalidTransaction(
+                    InvalidTransactionError::Eip4844Disabled,
+                )
+                .into())
+            }
+
+            self.evm_config.fill_tx_env(evm.tx_mut(), transaction, *sender);
+            if transaction.is_l1_message() {
+                evm.context.evm.env.cfg.disable_base_fee = true; // disable base fee for l1 msg
+            }
+
+            // RLP encode the transaction following eip 2718
+            let mut buf = BytesMut::with_capacity(transaction.encode_2718_len());
+            transaction.encode_2718(&mut buf);
+            let transaction_rlp_bytes = buf.freeze();
+            evm.context.evm.env.tx.scroll.rlp_bytes = Some(transaction_rlp_bytes.into());
+            evm.context.evm.env.tx.scroll.is_l1_msg = transaction.is_l1_message();
```

Comment:
Yes we need to fill those otherwise we will [error](https://github.com/scroll-tech/revm/blob/scroll-evm-executor/v49/crates/revm/src/scroll/handler_register.rs#L58-L62) during execution.

Reply:
But isn't that exclusively for the branch in which [`is_l1_message == false`](https://github.com/scroll-tech/revm/blob/91708073cefa4d22767d5a69a6636e68c09084e4/crates/revm/src/scroll/handler_register.rs#L53-L62). I think the branch for [`is_l1_messsage == true`](https://github.com/scroll-tech/revm/blob/91708073cefa4d22767d5a69a6636e68c09084e4/crates/revm/src/scroll/handler_register.rs#L81-L92) we don't use `.rlp_bytes` however I could be missing something.

Reply:
you might be right, will check the the fork and update if needed